### PR TITLE
Event manager

### PR DESCRIPTION
**Pull request by Sybsuper**
Events!!!!

### CODE REVIEW

File: world/location.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Location:
+	def __init__(self, x: float, y: float):
+		self.x = x
+		self.y = y
+
+	def get_x(self):
```
Comment (jubiman) [inline world/location.py]: Useless? Plis remove/give reason not to

File: events/playermoveevent.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	Example event
+	"""
+
+	def __init__(self, player, fromloc: Location, toloc: Location):
+		super(PlayerEvent, self).__init__(player)
+		super(CancelableEvent, self).__init__()
+		self.fromloc = fromloc
+		self.toloc = toloc
+
+	def get_from(self):
```
Comment (jubiman) [inline events/playermoveevent.py]: Useless? Plis remove/give reason not to

File: events/playermoveevent.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		return self.fromloc
+
+	def get_to(self):
```
Comment (jubiman) [inline events/playermoveevent.py]: Useless? Plis remove/give reason not to

File: events/playermoveevent.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+		return self.toloc
+
+	def set_to(self, toloc: Location):
```
Comment (jubiman) [inline events/playermoveevent.py]: Useless? Plis remove/give reason not to

File: events/playerevent.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from events.event import Event
+
+
+class PlayerEvent(Event):
+	def __init__(self, player):
+		self.player = player
+
+	def get_player(self):
```
Comment (jubiman) [inline events/playerevent.py]: Useless? Plis remove/give reason not to

File: events/cancelableevent.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+class CancelableEvent(Event):
+	"""
+	These events can be canceled.
+	"""
+
+	def __init__(self, canceled=False):
+		super().__init__()
+		self.canceled = canceled
+
+	def is_canceled(self):
```
Comment (jubiman) [inline events/cancelableevent.py]: Useless? Plis remove/give reason not to

File: events/cancelableevent.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+		return self.canceled
+
+	def set_canceled(self, canceled: bool):
```
Comment (jubiman) [inline events/cancelableevent.py]: Useless? Plis remove/give reason not to

**Review by jubiman — CHANGES_REQUESTED** [review]
See comments

**Comment by Sybsuper** [comment]
> See comments

Uhhh, too much Java I guess.

**Comment by jubiman** [comment]
> > See comments
> 
> Uhhh, too much Java I guess.

XD

**Review by jubiman — APPROVED** [review]
Koel

**Pull request merged**